plugin/evm/ftso_system_test.go

```diff
@@ -126,6 +127,283 @@ func TestFTSOSystem_Contracts(t *testing.T) {
+		require.EqualError(t, err, `unknown power block (height: 500)`)
+	})
+
+	t.Run("handles unknown current block error", func(t *testing.T) {
+		auth, be := simulatedBlockchain(t)
+		defer be.Close()
+
+		vpInt := big.NewInt(100)
+
+		unclaimedReward := big.NewInt(200)
+
+		rewardEpochDurationSeconds := big.NewInt(301)
+		rewardEpochPowerHeight := big.NewInt(5)
+		rewardEpochStartHeight := big.NewInt(400)
+		rewardEpochStartTime := big.NewInt(304)
+		rewardEpochsStartTs := big.NewInt(1) // this should be less than header.Time()
+
+		index1 := big.NewInt(123)
+		index2 := big.NewInt(321)
+		supportedIndices := []*big.Int{index1, index2}
+
+		indices := []*big.Int{big.NewInt(401)}
+		priceProvidersAddress1 := common.BigToAddress(big.NewInt(402))
+		priceProvidersAddresses := []common.Address{priceProvidersAddress1}
+
+		providerAddresses := []common.Address{common.BigToAddress(big.NewInt(500))}
+		nodes := [][20]byte{{1, 2}}
+
+		// VotePower contract
+		votePowerAddr := deployTestContract(t, auth, be, testAbiVotePower, testAbiVotePowerBin, vpInt)
+
+		// Wnat contract
+		wnatAddr := deployTestContract(t, auth, be, testAbiWnat, testAbiWnatBin, votePowerAddr)
+
+		// Reward contract
+		rewardAddr := deployTestContract(t, auth, be, testAbiReward, testAbiRewardBin, wnatAddr, unclaimedReward)
+
+		// Manager contract
+		ftsoManagerAddr := deployTestContract(t, auth, be, testAbiManager, testAbiManagerBin, rewardAddr, rewardEpochDurationSeconds, rewardEpochPowerHeight, rewardEpochStartHeight, rewardEpochStartTime, rewardEpochsStartTs)
+
+		// Registry contract
+		ftsoRegistryAddr := deployTestContract(t, auth, be, testAbiFtsoRegistry, testAbiFtsoRegistryBin, supportedIndices)
+
+		// Whitelist contract
+		voterWhitelisterAddr := deployTestContract(t, auth, be, testAbiVoterWhitelister, testAbiVoterWhitelisterBin, indices, priceProvidersAddresses)
+
+		// Validation contract
+		validationAddr := deployTestContract(t, auth, be, testAbiValidation, testAbiValidationBin, providerAddresses, nodes)
+
+		// Submitter contract
+		submitterAddr := deployTestContract(t, auth, be, testAbiSubmitter, testAbiSubmitterBin, voterWhitelisterAddr, ftsoRegistryAddr, ftsoManagerAddr)
+
+		be.Commit(true)
+
+		ftsoSystem := testFTSOSystem(t, be, submitterAddr, validationAddr)
+
+		epoch := big.NewInt(1).Uint64()
+
+		// crate four additional headers
+		for i := 0; i < 4; i++ {
+			be.Commit(true)
+		}
+
+		_, err := ftsoSystem.Snapshot(epoch)
+		require.EqualError(t, err, `unknown current block (height: 400)`)
+	})
+}
+
+func TestNewFTSOSystem(t *testing.T) {
+	_, be := simulatedBlockchain(t)
+	defer be.Close()
+
+	addressSubmitter := common.BigToAddress(big.NewInt(1))
+	addressValidation := common.BigToAddress(big.NewInt(2))
+
+	ftsoSystem, err := NewFTSOSystem(be.Blockchain(), addressSubmitter, addressValidation)
+	require.NoError(t, err)
+	assert.Equal(t, addressSubmitter, ftsoSystem.submitter.address)
+	assert.Equal(t, addressValidation, ftsoSystem.validation.address)
+}
```

Comment:
This should also have cases to test the situations where NewFTSOSystem fails. Though it's not really relevant for our deadline, it doesn't cost much to do IMO and it's mostly a lot of copy pasting to make sure that if some components are missing or invalid we don't mistakenly continue and silently fail.